src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/zip/ZipStepExecution.java

```diff
@@ -112,16 +112,19 @@ protected Void run() throws Exception {
     static class ZipItFileCallable extends MasterToSlaveFileCallable<Integer> {
         final FilePath zipFile;
         final String glob;
+        private final String zipFileName;
 
-        public ZipItFileCallable(FilePath zipFile, String glob) {
+        public ZipItFileCallable(FilePath zipFile, String glob, String zipFileName) {
             this.zipFile = zipFile;
             this.glob = StringUtils.isBlank(glob) ? "**/*" : glob;
+            this.zipFileName = zipFileName;
```

Comment:
what if you are writing a zip file in `/foo/bar/wibble/zip.zip` and you are archiving `/bob/jane/freddy` which contains a file called `zip.zip` - but not the same `zip.zip` as you are writing to?